src/rust-2024/rpit-lifetime-capture.md

```diff
@@ -1,12 +1,313 @@
+fn f_elided<T, const C: usize>() -> impl Sized {}
+//                                  ~~~~~~~~~~
+//                       The `use<..>` bound is elided here.
+//
+// In all editions, the above is equivalent to:
+fn f_explicit<T, const C: usize>() -> impl Sized + use<T, C> {}
+```
+
+In Rust 2021 and earlier editions, generic lifetime parameters are *not* included in this automatically-inserted `use<..>` bound for RPIT opaque types in the signature of bare functions and associated functions and methods within inherent impls.  However, starting in Rust 2024, these in-scope generic lifetime parameters *are* included.  E.g.:
+
+```rust
+# #![feature(precise_capturing)]
+fn f_elided(_: &()) -> impl Sized {}
+// In Rust 2021 and earlier, the above is equivalent to:
+fn f_2021(_: &()) -> impl Sized + use<> {}
+// In Rust 2024 and later, it's equivalent to:
+fn f_2024(_: &()) -> impl Sized + use<'_> {}
+```
+
+This makes the behavior consistent with RPIT opaque types in the signature of associated functions and methods within trait impls, uses of RPIT within trait definitions (RPITIT), and opaque `Future` types created by `async fn`, all of which capture all in-scope generic lifetime parameters by default in all editions.
+
+### Outer generic parameters
+
+When we say that an RPIT opaque type captures all in-scope generic parameters, note that this includes generic parameters from an outer impl.  E.g.:
+
+```rust
+# #![feature(precise_capturing)]
+struct S<T, const C: usize>((T, [(); C]));
+impl<T, const C: usize> S<T, C> {
+//   ~~~~~~~~~~~~~~~~~
+// These generic parameters are in scope.
+    fn f_elided<U>() -> impl Sized {}
+    //          ~       ~~~~~~~~~~
+    //          ^ This generic is in scope too.
+    //                  ^
+    //                  |
+    //   The `use<..>` bound is elided here.
+    //
+    // In all editions, it's equivalent to:
+    fn f_explicit<U>() -> impl Sized + use<T, U, C> {}
+}
+```
+
+### Lifetimes from higher-ranked binders
+
+Similarly, generic lifetime parameters introduced into scope by a higher-ranked `for<..>` binder are considered to be in scope.  E.g.:
+
+```rust
+# #![feature(precise_capturing)]
+trait Tr<'a> { type Ty; }
+impl Tr<'_> for () { type Ty = (); }
+
+fn f_elided() -> impl for<'a> Tr<'a, Ty = impl Copy + use<>> {}
+// In Rust 2021 and earlier, the above is equivalent to:
+fn f_2021() -> impl for<'a> Tr<'a, Ty = impl Copy + use<>> {}
+// In Rust 2024 and later, it's equivalent to:
+//fn f_2024() -> impl for<'a> Tr<'a, Ty = impl Copy + use<'a>> {}
+//                                        ~~~~~~~~~~~~~~~~~~~~
+// However, note that the capturing of higher-ranked lifetimes in
+// nested opaque types is not yet supported.
+```
+
+### Argument position impl Trait (APIT)
+
+Anonymous (i.e. unnamed) generic parameters created by the use of APIT (argument position impl Trait) are considered to be in scope when inserting an explicit `use<..>` bound for an elided one.  E.g.:
```

Comment:
```suggestion
Anonymous (i.e. unnamed) generic parameters created by the use of APIT (argument position impl Trait) are considered to be in scope for the purposes of capturing.  E.g.:
```

or something like that.

Reply:
All descriptions of automatically inserting `use<..>` bounds have been removed.